iothub/device/src/Transport/Amqp/AmqpUnit.cs

```diff
@@ -181,8 +180,9 @@ private void Cleanup()
             _amqpAuthenticationRefresher?.StopLoop();
             if (!_deviceIdentity.IsPooling())
             {
-                _amqpConnectionHolder?.Dispose();
+                _amqpConnectionHolder?.Shutdown();
```

Comment:
Can it instead call RemoveAmqpUnit?

Reply:
To avoid to call upper layer(cycle dependency).